src/include/processor/operator/copy_to/copy_to.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "common/copier_config/copier_config.h"
+#include "processor/operator/physical_operator.h"
+#include "common/task_system/task_scheduler.h"
+#include "processor/result/result_set.h"
+
+namespace kuzu {
+namespace processor {
+
+class CopyTo : public PhysicalOperator {
+public:
+    CopyTo(const common::CopyDescription& copyDescription, uint32_t id,
+        const std::string& paramsString)
+        : PhysicalOperator{PhysicalOperatorType::COPY_TO, id, paramsString}, copyDescription{copyDescription} {}
+
+    bool getNextTuplesInternal(ExecutionContext* context) override {
+        throw common::InternalException(
+            "to be implemented?");
+    }
+
+    std::string execute(common::TaskScheduler* taskScheduler, ExecutionContext* executionContext);
+
+//    void initLocalStateInternal(ResultSet* resultSet, ExecutionContext* context) override;
+
+    std::unique_ptr<PhysicalOperator> clone() override {
+        return make_unique<CopyTo>(copyDescription, id, paramsString);
+    }
+
+private:
+    common::CopyDescription copyDescription;
```

Comment:
Apart from copyDescription, you also need what vectors are you going to copy to. Which means you need 2 things

- vector<DataPos> vectorsToCopyPos
   -  You should get this from the constructor 
- vector<ValueVector> vectorsToCopy
   -  Initialize this through initLocalStateInternal

You can take a look at `payloads` in `ResultCollector` as a reference